### PR TITLE
Add support for groups in jwt authenticator

### PR DESCRIPTION
**Pull request by palash-goel**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticatorConfig.java [hunk 1]
```diff
@@ -84,6 +85,18 @@ public JwtAuthenticatorConfig setPrincipalField(String principalField)
         return this;
     }
 
+    public String getGroupsField()
+    {
+        return groupsField;
+    }
+
+    @Config("http-server.authentication.jwt.groups-field")
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticatorConfig.java:93]: nit. Add `@ConfigDescription`

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java [hunk 2]
```diff
@@ -56,11 +60,20 @@ public JwtAuthenticator(JwtAuthenticatorConfig config, SigningKeyResolver signin
...
-                .getBody()
-                .get(principalField, String.class))
+                        .getBody()
+                        .get(principalField, String.class))
                 .map(BasicPrincipal::new);
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    protected Set<String> extractGroupsFromToken(String token)
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java:70]: How about opaque tokens?

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticatorConfig.java [hunk 3]
```diff
@@ -84,6 +85,18 @@ public JwtAuthenticatorConfig setPrincipalField(String principalField)
         return this;
     }
 
+    public String getGroupsField()
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticatorConfig.java:88]: return `Optional<String>` and add `@NonNull` annotation to the method

File: core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java [hunk 4]
```diff
@@ -33,12 +35,14 @@
 {
     private final JwtParser jwtParser;
     private final String principalField;
+    private final String groupsField;
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/jwt/JwtAuthenticator.java:38]: use Optional or add `@Nullable annotation

File: core/trino-main/src/main/java/io/trino/server/security/AbstractBearerAuthenticator.java [hunk 5]
```diff
@@ -19,8 +19,10 @@
 import javax.ws.rs.container.ContainerRequestContext;
```
Comment (kokosing) [inline core/trino-main/src/main/java/io/trino/server/security/AbstractBearerAuthenticator.java:19]: Please add a test to `io.trino.server.security.oauth2.TestOAuth2WebUiAuthenticationFilter`

Comment (kokosing) [reply core/trino-main/src/main/java/io/trino/server/security/AbstractBearerAuthenticator.java:19]: see https://github.com/trinodb/trino/pull/9524

**Comment by mosabua** [comment]
:wave: @palash-goel  - this PR is inactive and doesn't seem to be under development. If you'd like to continue work on this at any point in the future, feel free to re-open.

**Pull request closed without merging**